health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java

```diff
@@ -298,4 +306,60 @@ public void validatePlanConfigExistence(PlanConfigurationRequest request) {
             throw new CustomException(INVALID_PLAN_CONFIG_ID_CODE, INVALID_PLAN_CONFIG_ID_MESSAGE);
         }
     }
+    
+    /**
+     * Validates the operations input against the Master Data Management System (MDMS) data.
+     *
+     * @param request  The PlanConfigurationRequest containing the plan configuration and other details.
+     * @param mdmsData The MDMS data containing the master rule configure inputs.
+     */
+    public void validateResourceMappingAgainstMDMS(PlanConfigurationRequest request, Object mdmsData) {
+        PlanConfiguration planConfiguration = request.getPlanConfiguration();
+        List<File> files = planConfiguration.getFiles();
+        List<String> templateIds = files.stream()
+                .map(File::getTemplateIdentifier)
+                .collect(Collectors.toList());
+        List<String> inputFileTypes = files.stream()
+                .map(File::getInputFileType)
+                .map(File.InputFileTypeEnum::toString)
+                .collect(Collectors.toList());
+
+        final String jsonPathForRuleInputs = "$." + MDMS_PLAN_MODULE_NAME + "." + MDMS_MASTER_SCHEMAS;
+        List<Object> ruleInputsListFromMDMS = null;
+        try {
+            log.info(jsonPathForRuleInputs);
+            ruleInputsListFromMDMS = JsonPath.read(mdmsData, jsonPathForRuleInputs);
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            throw new CustomException(JSONPATH_ERROR_CODE, JSONPATH_ERROR_MESSAGE);
+        }
+        List<String> allowedColumns = getIsTrueColoumnFromSchema(ruleInputsListFromMDMS, templateIds, inputFileTypes);
+        List<ResourceMapping> resourceMapping1 =  planConfiguration.getResourceMapping();  
+        if(allowedColumns.contains(ServiceConstants.BOUNDARY_CODE)) {
+        	Stream<ResourceMapping> d = resourceMapping1.stream().filter(f-> f.getMappedTo().equals(ServiceConstants.BOUNDARY_CODE));
+        if(!(d.count()>0)) {        	
+            throw new CustomException(REQUIRED_MAPPING_NOT_FOUND_CODE, REQUIRED_MAPPING_NOT_FOUND_MESSAGE);
+        }
+        }
+    }
+    
+    public static List<String> getIsTrueColoumnFromSchema(List<Object> schemas, List<String> templateIds, List<String> inputFileTypes) {
```

Comment:
It should be getIsTruePropertyFromSchema instead of getIsTrueColoumnFromSchema

Reply:
Done